src/imitation/policies/exploration_wrapper.py

```diff
@@ -0,0 +1,64 @@
+"""Wrapper to turn a policy into a more exploratory version."""
+
+from typing import Optional
+
+import numpy as np
+from stable_baselines3.common import vec_env
+
+from imitation.data import rollout
+
+
+class ExplorationWrapper:
+    """Wraps a PolicyCallable to create a partially randomized version.
+
+    This wrapper randomly switches between two policies: the wrapped policy,
+    and a uniformly random one. After each action, the current policy is kept
+    with a certain probability. Otherwise, one of these two policies is chosen
+    at random (without any dependence on what the current policy is).
+    """
+
+    def __init__(
+        self,
+        policy: rollout.PolicyCallable,
+        venv: vec_env.VecEnv,
+        random_prob: float,
+        stay_prob: float,
+        seed: Optional[int] = None,
+    ):
+        """Initializes the ExplorationWrapper.
+
+        Args:
+            policy: The policy to randomize.
+            venv: The environment to use (needed for sampling random actions).
+            random_prob: The probability of picking the random policy when switching.
+            stay_prob: The probability of staying with the current policy.
+            seed: The random seed to use.
+        """
+        self.wrapped_policy = policy
+        self.random_prob = random_prob
+        self.stay_prob = stay_prob
+        self.venv = venv
+
+        self.rng = np.random.RandomState(seed)
+        self.venv.action_space.seed(seed)
+
+        self.current_policy = policy
+        # Choose the initial policy at random
+        self._switch()
+
+    def _random_policy(self, states):
```

Comment:
```suggestion
    def _random_policy(self, states: np.ndarray) -> np.ndarray:
```
Is it states or obs?

Reply:
I went with `obs` now, I think `imitation` uses that (almost?) everywhere else